[opencolorio] implicitly depends on other library

### Is your feature request related to a problem? Please describe.

 [opencolorio] fails to build until I manually add [lcms] to my project vcpkg.json

### Proposed solution

Explicitly add [lcms] library to [opencolorio] dependencies.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

Hi @FrankXie05 

I've prepared a minimalistic repro here: https://github.com/AlexeySmolenchuk/vcpkg_repro_38153


Clone, checkout **main** branch, configure cmake.
```
git clone git@github.com:AlexeySmolenchuk/vcpkg_repro_38153.git --recursive
cd vcpkg_repro_38153
git checkout main
mkdir build
cd build
cmake ..
```
After bootstrapping I'm getting this error:
```
[cmake] CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
[cmake]   Could not find a package configuration file provided by "lcms2" with any of
[cmake]   the following names:
[cmake] 
[cmake]     lcms2Config.cmake
[cmake]     lcms2-config.cmake
...
-- Configuring incomplete, errors occurred!
```

Checkout to **workaround** branch, which explicitly set [lcms] as dependency. https://github.com/AlexeySmolenchuk/vcpkg_repro_38153/blob/workaround/vcpkg.json#L3 \
Configure cmake.
```
git checkout workaround
cmake ..
```
Now it's working.

Details:
OS Windows 10\
cmake-3.27

Thanks,
Alexey

--- a/ports/opencolorio/fix-dependency.patch
+++ b/ports/opencolorio/fix-dependency.patch
@@ -49,7 +49,7 @@ diff --git a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
 index 6a4932a..2a08f5a 100644
 --- a/src/cmake/Config.cmake.in
 +++ b/src/cmake/Config.cmake.in
-@@ -1,6 +1,12 @@
+@@ -1,6 +1,11 @@
  @PACKAGE_INIT@
 
  include(CMakeFindDependencyMacro)
@@ -58,7 +58,6 @@ index 6a4932a..2a08f5a 100644
 +find_dependency(pystring CONFIG)
 +find_dependency(yaml-cpp CONFIG)
 +find_dependency(minizip-ng CONFIG)
-+find_dependency(lcms2 CONFIG)
 
  if (NOT @BUILD_SHARED_LIBS@) # NOT @BUILD_SHARED_LIBS@
      if (APPLE)

--- a/ports/opencolorio/vcpkg.json
+++ b/ports/opencolorio/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "opencolorio",
   "version-semver": "2.2.1",
-  "port-version": 1,
+  "port-version": 2,
   "description": "OpenColorIO (OCIO) is a complete color management solution geared towards motion picture production with an emphasis on visual effects and computer animation. OCIO provides a straightforward and consistent user experience across all supporting applications while allowing for sophisticated back-end configuration options suitable for high-end production usage. OCIO is compatible with the Academy Color Encoding Specification (ACES) and is LUT-format agnostic, supporting many popular formats.",
   "homepage": "https://opencolorio.org/",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6426,7 +6426,7 @@
     },
     "opencolorio": {
       "baseline": "2.2.1",
-      "port-version": 1
+      "port-version": 2
     },
     "opencsg": {
       "baseline": "1.6.0",

--- a/versions/o-/opencolorio.json
+++ b/versions/o-/opencolorio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "07ef1b1be3450c82b92fa80f767cc04555fe88b4",
+      "version-semver": "2.2.1",
+      "port-version": 2
+    },
+    {
       "git-tree": "661b43b1d2c452e40476bbb64bb6e68dfc437c7b",
       "version-semver": "2.2.1",
       "port-version": 1

